File: backend/enum-jpa-validation-app/src/main/java/com/example/enumapp/web/validation/NoNullElementsValidator.java
package com.example.enumapp.web.validation;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

import java.util.Collection;

public class NoNullElementsValidator implements ConstraintValidator<NoNullElements, Collection<?>> {

    @Override
    public boolean isValid(Collection<?> value, ConstraintValidatorContext context) {
        if (value == null) {
            return true; // null list 는 @NotNull / @NotEmpty 가 담당
        }
        return value.stream().noneMatch(e -> e == null);
    }
}

File: backend/enum-jpa-querydsl-kotlin-app/src/main/kotlin/com/example/enumapp/web/validation/NoNullElementsValidator.kt
package com.example.enumapp.web.validation

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class NoNullElementsValidator : ConstraintValidator<NoNullElements, Collection<*>> {

    override fun isValid(value: Collection<*>?, context: ConstraintValidatorContext): Boolean {
        if (value == null) {
            return true // null list 는 @NotNull / @NotEmpty 가 담당
        }
        return value.none { it == null }
    }
}
